PhysicsTools/NanoAOD/python/custom_muon_cff.py

```diff
@@ -0,0 +1,276 @@
+    proc.muonTable.variables.jetDF = Var("?userCand('jetForLepJetVar').isNonnull()?max(userCand('jetForLepJetVar').bDiscriminator('pfDeepFlavourJetTags:probbb')+userCand('jetForLepJetVar').bDiscriminator('pfDeepFlavourJetTags:probb')+userCand('jetForLepJetVar').bDiscriminator('pfDeepFlavourJetTags:problepb'),0.0):0.0",float,doc="b-tagging discriminator of the jet matched to the lepton, for muon MVA")
+    proc.muonTable.variables.jetCSVv2 = Var("?userCand('jetForLepJetVar').isNonnull()?max(userCand('jetForLepJetVar').bDiscriminator('pfCombinedSecondaryVertexV2BJetTags'),0.0):0.0",float,doc="CSVv2 b-tagging discriminator of the jet matched to the lepton, for muon MVA")
+    
+    #Dxy Dz variables as of Spark tool
+    proc.muonTable.variables.innerTrackDxy = Var("? userInt('isGoodVertex') ? userFloat('innerTrackDxy') : -99.9",float,doc = "dxy from Primary Vertex calculated with Inner Track")
+    proc.muonTable.variables.innerTrackDz = Var("? userInt('isGoodVertex') ? userFloat('innerTrackDz') : -99.9",float,doc= "dz from Primary Vertex calculated with Inner Track")
+
+    #nSegments
+    proc.muonTable.variables.nsegments = Var("userInt('nsegments')", int, doc = "nsegments as of Spark-tool")
+
+    #Sim Variables
+    proc.muonTable.variables.simType = Var("? simType() ? simType() : -99",int,doc="simType")
+    proc.muonTable.variables.simExtType = Var("? simExtType() ? simExtType() : -99",int,doc="simExtType")
+    proc.muonTable.variables.simFlavour = Var("? simFlavour() ? simFlavour() : -99",int,doc="simFlavour")
+    proc.muonTable.variables.simHeaviestMotherFlavour = Var(" ? simHeaviestMotherFlavour() ? simHeaviestMotherFlavour() : -99",int,doc="simHeaviestMotherFlavour")
+    proc.muonTable.variables.simPdgId = Var("? simPdgId() ? simPdgId() : -99",int,doc="simPdgId")
+    proc.muonTable.variables.simMotherPdgId = Var("? simMotherPdgId() ? simMotherPdgId() : -99",int,doc="simMotherPdgId")
+    proc.muonTable.variables.simBX = Var("? simBX() ? simBX() : -99",int,doc="simBX")
+    proc.muonTable.variables.simProdRho = Var("? simProdRho() ? simProdRho(): -99",float,doc="simProdRho")
+    proc.muonTable.variables.simProdZ = Var("? simProdZ() ? simProdZ(): -99",float,doc="simProdZ")
+    proc.muonTable.variables.simPt = Var("? simPt() ? simPt(): -99",float,doc="simPt")
+    proc.muonTable.variables.simEta = Var("? simEta() ? simEta(): -99",float,doc="simEta")
+    proc.muonTable.variables.simPhi = Var("? simPhi() ? simPhi(): -99",float,doc='simPhi')
+    
+    
+    return proc
+
+
+def PrepMuonCustomNanoAOD(process, runOnMC):
+    
+    process = AddPFTracks(process)
+    process = AddVariablesForMuon(process)
+    #process = ChangeNanoAODContent(process)
+
+    return process
+
+
+
+def PrepMuonCustomNanoAOD_MC(process):
```

Comment:
Likewise in the current configuration it looks like data and MC can use the same customization function (which is anyway preferred)